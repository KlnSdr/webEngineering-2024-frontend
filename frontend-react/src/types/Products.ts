interface NeededProduct {
  id: number; // needed because react decided to be difficult today!
  productName: string;
  amount: number;
}

interface Product {
    name: string;
    unit: string;
}

export type { NeededProduct, Product };
