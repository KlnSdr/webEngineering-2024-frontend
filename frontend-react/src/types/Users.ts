interface User {
    userId: number;
    userName: string;
}

export type { User };
