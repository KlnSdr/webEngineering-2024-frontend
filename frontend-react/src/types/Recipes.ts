import { NeededProduct } from "./Products";

interface CreateRecipe {
  title: string;
  image: File | null;
  description: string;
  products: NeededProduct[];
}

export type { CreateRecipe };
