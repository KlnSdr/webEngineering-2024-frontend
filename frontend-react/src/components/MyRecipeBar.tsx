import React from "react";
import {Recipe} from "../types/Recipes";
import {Heading2} from "./Heading";

interface MyRecipeBarProps {
    Recipe: Recipe;

}

const MyRecipeBar: React.FC<MyRecipeBarProps> = ({Recipe}) => {
    return (
            <div className="TextArea">
                <Heading2 headingText={Recipe.title}/>
            </div>
    );
};

export default MyRecipeBar;

