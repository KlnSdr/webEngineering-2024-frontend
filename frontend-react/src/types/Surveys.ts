import {Recipe} from "./Recipes";

interface Survey{
    id: number;
    title: string;
    participants: string[];
    creator: string;
    recipeVote: {[key: string]: number};
    options: string[];
    creationDate: Date;
}

interface UpdateSurvey {
    id: number;
    title: string;
    participants: string[];
    creator: string;
    recipeVote: {[key: string]: number};
    options: Recipe[];
    creationDate: Date;
}

interface CreateSurvey {
    title: string;
    options: Recipe[];
}

export type {Survey, CreateSurvey,UpdateSurvey};

