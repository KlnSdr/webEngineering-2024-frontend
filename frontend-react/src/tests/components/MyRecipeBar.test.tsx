import React from "react";
import { render, screen } from "@testing-library/react";
import '@testing-library/jest-dom/extend-expect';
import MyRecipeBar from "../../components/MyRecipeBar";
import {CreateRecipe, Recipe} from "../../types/Recipes";

describe("MyRecipeBar Component", () => {
    const recipe: Recipe = {
        id: 1,
        title: "Test Recipe",
        description: "Test Description",
        imgUri: "test.jpg",
        isPrivate: false,
        creationDate: new Date(),
        ownerUri: "test",
        likedByUserUris: [],
        products:[]


    };

    test("renders with correct recipe title", () => {
        const {asFragment} = render(<MyRecipeBar Recipe={recipe} />);

        const titleElement = screen.getByRole("heading", { level: 2 });
        expect(titleElement).toHaveTextContent(recipe.title);
        expect(asFragment()).toMatchSnapshot();
    });
});
