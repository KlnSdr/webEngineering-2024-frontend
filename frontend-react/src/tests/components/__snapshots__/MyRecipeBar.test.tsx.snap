// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyRecipeBar Component renders with correct recipe title 1`] = `
<DocumentFragment>
  <div
    class="TextArea"
  >
    <h2
      class="FoodTitle"
    >
      Test Recipe
    </h2>
  </div>
</DocumentFragment>
`;
